types/api/api.go

```diff
@@ -50,9 +57,19 @@ type ListRequest struct{}
 type GetRequest struct {
 	// Name is the name of the secret to fetch.
 	Name string
-	// Version is the version to fetch, or SecretVersionDefault to let
-	// the server pick a version.
+
+	// Version is the version to fetch, or SecretVersionDefault to specify that
+	// the server should return the latest active version.
 	Version SecretVersion
+
+	// UpdateIfChanged, if true, instructs the server to return the latest
+	// active version of the secret if (and only if) the latest active version
+	// is different from Version. If the latest active version is equal to
+	// Version, the server reports 304 Not Modified and returns no value.
+	//
+	// If Version == SecretVersionDefault, this flag is ignored and the latest
+	// active version is returned unconditionally.
+	UpdateIfChanged bool
```

Comment:
Hmm, I wonder if this would scan better is the new field were `HaveVersion SecretVersion`, rather than make Version's meaning change based on a sibling bool.

It's not great because a request with both Version and HaveVersion set is arguably nonsensical (albeit harmlessly so). I don't have a strong feeling here, this is also fine, just spitballing.

Reply:
I had the same thought, and I came to the (lightly-held) conclusion that it was more confusing to have _two_ versions than a version and a flag. The rationale being that we already have a special meaning for some values of Version, so adding more is not too bad.

Nonetheless, I am likewise not strongly attached, though I think this is probably a good default.